Website/Packages/datax-pipeline/src/modules/flowDefinition/components/schedule/oneTimeScheduleSettings.jsx

```diff
@@ -0,0 +1,234 @@
+// Licensed under the MIT License
+// *********************************************************************
+import React from 'react';
+import PropTypes from 'prop-types';
+import * as Helpers from '../../flowHelpers';
+import * as Models from '../../flowModels';
+import { TextField, Label, Dropdown } from 'office-ui-fabric-react';
+import DateTimePicker from 'react-datetime-picker';
+
+export default class OneTimeScheduleSettings extends React.Component {
+    constructor(props) {
+        super(props);
+    }
+
+    render() {
+        return (
+            <div style={rootStyle}>
+                {this.renderAlias()}
+                {this.renderTypeDisplayName()}
+                {this.renderInterval()}
+                {this.renderWindow()}
+                {this.renderTimeRange()}
+            </div>
+        );
+    }
+
+    renderAlias() {
+        return (
+            <div style={sectionStyle}>
+                <TextField
+                    className="ms-font-m info-settings-textbox"
+                    spellCheck={false}
+                    label="Alias"
+                    value={this.props.batch.id}
+                    onChange={(event, value) => this.props.onUpdateBatchName(value)}
+                    onGetErrorMessage={value => this.validateProperty(value)}
+                />
+            </div>
+        );
+    }
+    
+    renderTypeDisplayName() {
+        return (
+            <div style={batchTypeSection}>
+                <TextField
+                    className="ms-font-m info-settings-textbox"
+                    spellCheck={false}
+                    label="Batch Type"
+                    disabled={true}
+                    value={this.props.batchTypeDisplayName}
+                />
+            </div>
+        );
+    }
+
+    renderTimeRange() {
+        return (
+            <div >
+                {this.renderStartTime()}
+                {this.renderEndTime()}
+            </div>
+        );
+    }
+    
+    renderStartTime() {
+        return (
+            <div style={sectionStyle}>
+                <Label className="ms-font-m info-settings-textbox">Start Time</Label>
+                <DateTimePicker
+                    className="ms-font-m info-settings-textbox"
+                    value={this.props.batch.properties.startTime}
+                    onChange={(value) => this.props.onUpdateBatchStartTime(value)}
+                />
+            </div>
+        );
+    }
+
+    renderEndTime() {
+        return (
+            <div style={sectionStyle}>
+                <Label className="ms-font-m info-settings-textbox">End Time</Label>
+                <DateTimePicker
+                    className="ms-font-m info-settings-textbox"
+                    value={this.props.batch.properties.endTime}
+                    onChange={(value) => this.props.onUpdateBatchEndTime(value)}
+                />
+            </div>
+        );
+    }
+
+    renderInterval() {
```

Comment:
Looking at your shared screenshots of the UI, I don't like how you are rendering the width of the sections for Recurrence, Delay, Window (in Recurring Job Settings) and Interval, Window (in One Time Job Settings). If user has wide monitors and most do, you are rendering the text box and drop downs to take entire width of the settings panel. Lets not do this and fix the width. For example, in all cases for the text box, you simply accept a number and in all cases, the Unit drop down is just a short text, I I think you can make the width of these section container style reuse the same width specified by Alias, Batch Type, Start Time, End Time, etc. This way both controls will take up 50% of width where the width is consitent with the other controls in the UI.

Reply:
please make sure you do this for both the one time and recurring job settings.

Reply:
done. use the same width.